components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/LegacyJobOrchestrator.cs

```diff
@@ -1054,15 +1077,24 @@ IJsonSerializer jsonSerializer
         }
 
         /// <inheritdoc/>
-        public async Task<PublishedNodesEntryModel> GetDiagnosticInfoAsync(
-            PublishedNodesEntryModel request,
+        public async Task<List<JobDiagnosticInfoModel>> GetDiagnosticInfoAsync(
             CancellationToken ct = default) {
             _logger.Information("{nameof} method triggered", nameof(GetDiagnosticInfoAsync));
+            var sw = Stopwatch.StartNew();
             await _lockConfig.WaitAsync(ct).ConfigureAwait(false);
```

Comment:
>_lockConfig [](http://example.com/codeflow?start=18&length=11)

let's use the _lockJobs here. the _publisherDiagnosticInfo is not related the the _lockConfig

Reply:
changed